### PR TITLE
Script that checks that all test oracles are compilable

### PR DESCRIPTION
**Pull request by kelloggm**
I expect CI to fail. @LoiNguyenCS you should check the failures here and then make PRs that correct the bugs that it exposes. Once CI on this PR is green, we can merge it and then enforce this requirement for tests going forward.

### CODE REVIEW

**Comment by LoiNguyenCS** [comment]
Professor, I have a little concern about this script. 

This script appears to compile each program as an isolated component. For instance, consider a scenario with two files: `com/example/Child.java` and `sample/pack/MyType.java`. Let's assume that the `Child` class imports the `MyType` class. Attempting to compile `com/example/Child.java` independently may result in an error because javac lacks information about the location of `MyType.java`.

**Comment by LoiNguyenCS** [comment]
I've figured out the problems. It was due to how Specimin handles import statements. If a class is deleted because it is not used by the target methods, Specimin should delete the import statements for that class in the final output too. Otherwise we've got an "unfound" error.

**Comment by LoiNguyenCS** [comment]
Professor,

I've made two changes on this branch. Specimin will now:
1. Remove import statements that import unused classes.
2. Try to use fully-qualified names to correct types, instead of simple names.

I am not sure why the CI is failing. Please take a look. Thank you.

**Comment by LoiNguyenCS** [comment]
When I tried running "typecheck_test_outputs.sh" in my local machine, I also got an error message without any further explanation.

**Comment by LoiNguyenCS** [comment]
Professor @kelloggm,

I have fixed the failure of CI.

Please take a look. Thank you.

**Review by LoiNguyenCS — APPROVED** [review]
LGTM. (since I wrote the codes for these latest commits)

File: src/main/java/org/checkerframework/specimin/GetTypesFullNameVisitor.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+   */
+  private Map<String, Set<String>> fileAndAssociatedTypes = new HashMap<>();
+
+  /**
+   * Get the map of files' directories and types used within those files.
+   *
+   * @return the value of fileAndAssociatedTypes
+   */
+  public Map<String, Set<String>> getFileAndAssociatedTypes() {
+    return fileAndAssociatedTypes;
```
Comment (kelloggm) [inline src/main/java/org/checkerframework/specimin/GetTypesFullNameVisitor.java:34]: This should be unmodifiable, using `Collections.unmodifiableMap` (or whatever that method is called).

**Review by kelloggm — COMMENTED** [review]
I'm concerned that your changes remove signature checker annotations. Given the timeline we're on, I'm okay with doing that for now. However, please open an issue indicating that there is a need to add more signature checker annotations on these classes, which we can return to after the ISSTA deadline.

Once you've done that and made the one other change I requested, this is ready to merge without another review.

**Pull request merged**